portal-impl/src/com/liferay/portal/service/impl/UserLocalServiceImpl.java

```diff
@@ -4515,6 +4519,26 @@ public void updateAsset(
 
 		Group companyGroup = company.getGroup();
 
+		if (assetCategoryIds == null) {
+			long classNameId = _classNameLocalService.getClassNameId(
+				User.class.getName());
+
+			for (AssetVocabulary assetVocabulary :
+					_assetVocabularyLocalService.getGroupVocabularies(
+						companyGroup.getGroupId())) {
+
+				if (assetVocabulary.isRequired(
+						classNameId, user.getUserId(),
+						companyGroup.getGroupId())) {
+
+					AssetEntry assetEntry = _assetEntryLocalService.getEntry(
+						companyGroup.getGroupId(), user.getUuid());
+
+					assetCategoryIds = assetEntry.getCategoryIds();
+				}
+			}
```

Comment:
is not possible to do the same with a Transformer?

Reply:
Since we're just adding it once, I didn't think a Transformer was appropriate. I did just remember to add a break though since technically we should only need to re-add the user's assetCategoryIds once as far as I know.